SHACL Overview, what's new

nicholascar opened the pull request:
* will satisfy the requirement to have information on what's new for Core & SPARQL Horizontal Review
* cribbed from LinkedIn and extended


Rendered at https://raw.githack.com/w3c/data-shapes/whats-new/shacl12-overview/index.html

Expected updates from Nick:

* ordering of table
* updating links to specs as they become real

nicholascar committed:
what's new, cribbed from LinkedIn and extended

HolgerKnublauch commented:
SHACL 1.1 was never published and was ever only in draft stage. If you want to refer to the old documents, it should be 1.0 (Core and SPARQL) for sure, but SHACL-AF could be referred to as 1.1.

Under what's new, I suggest to add "Misc New Constraint Components": sh:singleLine, sh:memberShape, sh:minListLength, sh:maxListLength, sh:uniqueMembers.

nicholascar commented:
Oh yes, I was sort of trialing out SHACL 1.1 instead of SHACL 1.0. The thinking being that SHACL was published without a version number but based on RDF 1.1. So if we are aligning current SHACL to RDF 1.2, we should retcon all previous SHACL docs to 1.1.

Obviously this thought wasn't what we originally discussed, but how does it strike you as a logic?

HolgerKnublauch commented:
The original https://www.w3.org/TR/shacl/ doesn't have any version number, so couldn't we just call them "Previous SHACL specifications", without any specific version number?

nicholascar commented:
> The original https://www.w3.org/TR/shacl/ doesn't have any version number, so couldn't we just call them "Previous SHACL specifications", without any specific version number?

Yes, we can do that. I'll apply that label across the board via updates to this PR so we are consistent.

TallTed commented on shacl12-overview/index.html:
```diff
@@ -373,61 +373,211 @@ <h2>Introduction</h2>
-				SHACL 1.0 Core <a href="https://www.w3.org/TR/2017/REC-shacl-20170720/#shacl-shacl">included an appendix</a> providing a shapes graph to be used to validate shapes graphs according to the SHACL 1.0 Core specification. SHACL 1.2 has moved this information from the Core specification to an appendix in this Overview to allow for information relevant to parts of SHACL 1.2 not in the Core. A note  ...
-			</p>
+      <p>
+        SHACL 1.2 caters for <a href="https://www.w3.org/TR/rdf12-new/">new developments in RDF 1.2 and SPARQL 1.2</a>, formalises versions of original SHACL documents published only as Notes and takes SHACL into new areas such as rules for inferencing and profile defining.
```
W3C documents typically use US English (e.g., `formalizes`). UK English (e.g., `formalises`) is _sometimes_ used, in which case each document should be internally consistent.
```suggestion
        SHACL 1.2 covers <a href="https://www.w3.org/TR/rdf12-new/">new developments in RDF 1.2 and SPARQL 1.2</a>, formalizes versions of original SHACL documents published only as Notes, and takes SHACL into new areas such as rules for inferencing and profile definition.
```

TallTed commented on shacl12-overview/index.html:
```diff
@@ -373,61 +373,211 @@ <h2>Introduction</h2>
+        SHACL 1.2 caters for <a href="https://www.w3.org/TR/rdf12-new/">new developments in RDF 1.2 and SPARQL 1.2</a>, formalises versions of original SHACL documents published only as Notes and takes SHACL into new areas such as rules for inferencing and profile defining.
+      </p>
+      <p>
+        The original SHACL specifications, published in 2015 and which we now call SHACL 1.1, are listed in the <a href="#shacl-1.1">SHACL 1.1</a> section below. SHACL 1.2's specifications are also listed below in the <a href="#shacl-1.2">SHACL 1.2</a> section.
```
```suggestion
        The original SHACL specifications, published in 2015 and now considered SHACL 1.1 as they were aligned with RDF 1.1, are listed in the <a href="#shacl-1.1">SHACL 1.1</a> section below. The SHACL 1.2 specifications are also listed below, in the <a href="#shacl-1.2">SHACL 1.2</a> section.
```

TallTed commented on shacl12-overview/index.html:
```diff
@@ -373,61 +373,211 @@ <h2>Introduction</h2>
+        The original SHACL specifications, published in 2015 and which we now call SHACL 1.1, are listed in the <a href="#shacl-1.1">SHACL 1.1</a> section below. SHACL 1.2's specifications are also listed below in the <a href="#shacl-1.2">SHACL 1.2</a> section.
+      </p>
+      <p>
+        following are the major new things in SHACL 1.2 and the specifications they appear in.
```
```suggestion
        Following are the major new things in SHACL 1.2 and the specifications in which they appear.
```

TallTed commented on shacl12-overview/index.html:
```diff
@@ -373,61 +373,211 @@ <h2>Introduction</h2>
+              Derived Properties
+            </td>
+            <td>
+              SHACL 1.1 shapes and constraints could only operate on asserted triples in a graph and inferencing was left as an optional pre-processing step, using languages like RDF Schema and OWL. SHACL 1.2 introduces its own inferencing and reasoning capabilities, making SHACL more self-contained, and covering different use cases than RDFS/OWL.
```
```suggestion
              In SHACL 1.1, shapes and constraints could only operate on asserted triples in a graph, and inferencing was left as an optional pre-processing step using languages like RDF Schema and OWL. SHACL 1.2 introduces its own inferencing and reasoning capabilities, which make SHACL more self-contained and cover different use cases than RDFS/OWL.
```

TallTed commented on shacl12-overview/index.html:
```diff
@@ -373,61 +373,211 @@ <h2>Introduction</h2>
+              Flexible Target Nodes
+            </td>
+            <td>
+              In SHACL 1.2, node expressions can be used in multiple places beside derived properties. In particular, they can also be used to compute the target nodes of a shape. This provides greater flexibility than the built-in target types such as <code>sh:targetClass</code> and <code>sh:targetSubjectsOf</code>.
```
```suggestion
              In SHACL 1.2, node expressions can be used in more places than derived properties, to which they were limited in SHACL 1.1. Particularly, they can now also be used to compute the target nodes of a shape. This provides greater flexibility than the built-in target types such as <code>sh:targetClass</code> and <code>sh:targetSubjectsOf</code>.
```

TallTed commented on shacl12-overview/index.html:
```diff
@@ -373,61 +373,211 @@ <h2>Introduction</h2>
+            </td>
+          </tr>
+          <tr>
+            <td>Inference Rules</td>
+            <td>
+              <p>
+                The new <em>Inferencing Rules</em> specification allows the representation and execution of <a href="https://en.wikipedia.org/wiki/Datalog">Datalog</a>-like inference rules which have a straight-forward mapping to SPARQL <code>CONSTRUCT</code> statements for execution, and integrate with other SHACL features such as node expressions.
```
It seems odd to title this side-spec `Inferencing Rules` when the things it discusses are generally referred to as `Inference Rules`. I suggest we change most if not all instances of the former, including the title of the other document, to the latter.
```suggestion
            <td>Inference Rules</td>
            <td>
              <p>
                The new <em>Inference Rules</em> specification allows the representation and execution of <a href="https://en.wikipedia.org/wiki/Datalog">Datalog</a>-like inference rules which have a straight-forward mapping to SPARQL <code>CONSTRUCT</code> statements for execution, and integrate with other SHACL features such as node expressions.
```

TallTed commented on shacl12-overview/index.html:
```diff
@@ -373,61 +373,211 @@ <h2>Introduction</h2>
+                The new <em>Inferencing Rules</em> specification allows the representation and execution of <a href="https://en.wikipedia.org/wiki/Datalog">Datalog</a>-like inference rules which have a straight-forward mapping to SPARQL <code>CONSTRUCT</code> statements for execution, and integrate with other SHACL features such as node expressions.
+              </p>
+              <p>
+                This work was inspired by the SPARQL-based rules introduced by the <a href="https://www.w3.org/TR/shacl-af/">SHACL 1.1 Advanced Features</a> note and older technologies in particular <a href="https://spinrdf.org/">SPIN</a>.
```
```suggestion
                This work was inspired by the SPARQL-based rules introduced by the <a href="https://www.w3.org/TR/shacl-af/">SHACL 1.1 Advanced Features</a> note (commonly referred to as "SHACL-AF") and older technologies such as <a href="https://spinrdf.org/">SPIN</a>.
```

TallTed commented on shacl12-overview/index.html:
```diff
@@ -373,61 +373,211 @@ <h2>Introduction</h2>
+              Better Syntax for Unions of Datatypes and Classes
+            </td>
+            <td>
+              In SHACL 1.1, when you wanted to express that the datatype of a property was either <code>xsd:string</code>, <code>rdf:langString</code> or <code>rdf:HTML</code>, you needed to use a verbose, repetitive construct. In SHACL 1.2, this can be written as a single list.
```
```suggestion
              In SHACL 1.1, when you wanted to express that the datatype of a property was either <code>xsd:string</code>, <code>rdf:langString</code>, or <code>rdf:HTML</code>, you needed to use a verbose, repetitive construct. In SHACL 1.2, this can be written as a single list.
```

TallTed commented on shacl12-overview/index.html:
```diff
@@ -373,61 +373,211 @@ <h2>Introduction</h2>
+              Constraints on RDF 1.2 Reification
+            </td>
+            <td>
+              The major new feature in RDF 1.2 is reification - RDF statements about other statements. SHACL 1.2 introduces new constraint properties <code>sh:reifierShape</code> and <code>sh:reificationRequired</code> which allows SHACL to chain a shape targeting a triple to another shape targeting reification elements.
```
```suggestion
              The major new feature in RDF 1.2 is reification, enabling RDF statements to be easily made about other RDF statements. SHACL 1.2 introduces new constraint properties (<code>sh:reifierShape</code> and <code>sh:reificationRequired</code>) which allow a shape targeting a triple to be chained to another shape targeting reification elements.
```

TallTed commented on shacl12-overview/index.html:
```diff
@@ -373,61 +373,211 @@ <h2>Introduction</h2>
+              Use of Reification in Constraint Definitions
+            </td>
+            <td>
+              In SHACL 1.1, the severity and messages of a constraint had to be declared for the surrounding shape sometimes requiring artificial intermediate shapes to be introduced only to change the severity or a message. SHACL 1.2 syntax is more flexible allowing severity and messages to be directly attached to individual constraint triples.
```
```suggestion
              In SHACL 1.1, the severity and messages of a constraint had to be declared for the surrounding shape, sometimes requiring artificial intermediate shapes to be introduced to change only the severity or a message. SHACL 1.2 syntax is more flexible, allowing severity and messages to be directly attached to individual constraint triples.
```

TallTed commented on shacl12-overview/index.html:
```diff
@@ -373,61 +373,211 @@ <h2>Introduction</h2>
+              The <code>sh:ShapeClass</code> Metaclass
+            </td>
+            <td>
+              A new class <code>sh:ShapeClass</code>, comparable to <code>owl:Class</code>, used to define classes that can declare constraints that apply to all instances of the class. It is equivalent to OWL syntax for declaring a class that may hold <a href="https://www.w3.org/TR/owl2-primer/#Annotating_Axioms_and_Entities">OWL axioms</a>. This allows SHACL to be a more self-contained ontology ...
```
```suggestion
              A new class, <code>sh:ShapeClass</code>, comparable to <code>owl:Class</code>, is used to define classes that can declare constraints that apply to all instances of the class. It is equivalent to OWL syntax for declaring a class that may hold <a href="https://www.w3.org/TR/owl2-primer/#Annotating_Axioms_and_Entities">OWL axioms</a>. This allows SHACL to be a more self-contained ontology modeling language.
```

TallTed commented on shacl12-overview/index.html:
```diff
@@ -373,61 +373,211 @@ <h2>Introduction</h2>
+              Cleaner Separation between Core and SPARQL
+            </td>
+            <td>
+              We now have a cleaner separation of Core and SPARQL concerns into separate specifications. This hopefully clarifies dependencies, indicates SHACL Core is not dependent on SPARQL and makes some implementations easier.
```
```suggestion
              We now have a cleaner separation of Core and SPARQL concerns into separate specifications. This helps indicate SHACL Core is not dependent on SPARQL, clarify other dependencies, and make some implementations easier.
```

TallTed commented on shacl12-overview/index.html:
```diff
@@ -373,61 +373,211 @@ <h2>Introduction</h2>
+            </td>
+            <td>
+              <p>
+                The SHACL Community Group, which formed after the publication of SHCAL 1.1, produced an informal <a href="https://w3c.github.io/shacl/shacl-compact-syntax/">SHACL Compact Syntax Report</a> which defines a compact syntax for core elements of SHACL 1.1.
```
```suggestion
                The SHACL Community Group, which formed after the publication of SHACL 1.1, produced an informal <a href="https://w3c.github.io/shacl/shacl-compact-syntax/">SHACL Compact Syntax Report</a> which defines a compact syntax (sometimes referred to as SHACL-C) for core elements of SHACL 1.1.
```

TallTed commented on shacl12-overview/index.html:
```diff
@@ -373,61 +373,211 @@ <h2>Introduction</h2>
+                The SHACL Community Group, which formed after the publication of SHCAL 1.1, produced an informal <a href="https://w3c.github.io/shacl/shacl-compact-syntax/">SHACL Compact Syntax Report</a> which defines a compact syntax for core elements of SHACL 1.1.
+              </p>
+              <p>
+                We now have a formalisation of that syntax with updates to aline with the other SHACL 1.2 specifications.
```
```suggestion
                We now have a formalization of that syntax with updates to aline with the other SHACL 1.2 specifications.
```

TallTed commented on shacl12-overview/index.html:
```diff
@@ -373,61 +373,211 @@ <h2>Introduction</h2>
+              </p>
+            </td>
+            <td>
+              <a href="https://www.w3.org/TR/shacl12-cs/">Compact Syntax</a>
```
The "Compact Syntax" document is now named `shacl12-cs`. I wonder whether we should try to change all references to `SHACL-C` to instead refer to `SHACL-CS`?

TallTed commented on shacl12-overview/index.html:
```diff
@@ -373,61 +373,211 @@ <h2>Introduction</h2>
+              A formal SHACL UI
+            </td>
+            <td>
+              SHACL 1.1 has been used in industry for user interface generation since publication with non-W3C extensions such as <a href="https://www.datashapes.org/">DASH</a>. This is now formalised in a new specification.
```
```suggestion
              Industry has used SHACL 1.1 for user interface generation since publication, with non-W3C extensions such as <a href="https://www.datashapes.org/">DASH</a>. This is now formalized in a new specification.
```

TallTed commented on shacl12-overview/index.html:
```diff
@@ -373,61 +373,211 @@ <h2>Introduction</h2>
+              Defined profiling mechanisms with SHACL
+            </td>
+            <td>
+              SHACL has always been used to <em>profile</em> RDF graphs and there are now profiles <em>of</em> SHACL - Core, SPARQL, UI etc. - so a new specification has been published to defin profiling mechanisms with SHACL and which aligns with other W3C specifications, in particular the <em>Profiles Vocabulary</em> [[DX-PROF]]
```
```suggestion
              Profiling mechanisms defined with SHACL
            </td>
            <td>
              SHACL has always been used to <em>profile</em> RDF graphs, and there are now profiles <em>of</em> SHACL — Core, SPARQL, UI, etc. — so a new specification has been published to define profiling mechanisms for use with SHACL, and to align with other W3C specifications, particularly the <em>Profiles Vocabulary</em> [[DX-PROF]].
```

TallTed commented on shacl12-overview/index.html:
```diff
@@ -373,61 +373,211 @@ <h2>Introduction</h2>
+          <dd>defines graph expressions used to determine focus nodes in SHACL</dd>
+          <dt><a href="https://w3c.github.io/data-shapes/shacl12-inf-rules/">SHACL 1.2 Inference Rules</a></dt>
+          <dd>defines SHACL's methods of rule-based inference</dd>
+          <dt><a href="https://w3c.github.io/data-shapes/shacl12-ui/">SHACL 1.2 UI</a></dt>
```
It seems odd to use the `UI` acronym here when all the other titles are fully spelled out.
```suggestion
          <dt><a href="https://w3c.github.io/data-shapes/shacl12-ui/">SHACL 1.2 User Interfaces</a></dt>
```

TallTed commented on shacl12-overview/index.html:
```diff
@@ -373,61 +373,211 @@ <h2>Introduction</h2>
+        <h3>
+          SHACL 1.1
+        </h3>
+        <p><em>The original SHACL specifications, now referred to as SHACL 1.1, are deprecated in favour of SHACL 1.2 specifications, above.</em></p>
```
```suggestion
        <p><em>The original SHACL specifications listed below, now referred to as SHACL 1.1, are deprecated in favor of the SHACL 1.2 specifications, listed above.</em></p>
```

TallTed reviewed (dismissed):
Grammar, punctuation, US English, etc.

ajnelson-nist committed:
Update shacl12-overview/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

ajnelson-nist committed:
Update shacl12-overview/index.html

Committing for grammar remarks, but this does not visit the topic raised about whether there officially exists a "SHACL 1.1".

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

ajnelson-nist committed:
Update shacl12-overview/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

ajnelson-nist committed:
Update shacl12-overview/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

ajnelson-nist committed:
Update shacl12-overview/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

ajnelson-nist committed:
Update shacl12-overview/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

ajnelson-nist committed:
Update shacl12-overview/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-overview/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-overview/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-overview/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-overview/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-overview/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-overview/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-overview/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-overview/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-overview/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

nicholascar committed:
Update shacl12-overview/index.html

Co-authored-by: Ted Thibodeau Jr <tthibodeau@openlinksw.com>

TallTed commented:
Note that you can use "add to batch" to make a single commit from multiple change requests. "Squash and merge" can also help to keep the commit history cleaner, while preserving contribution records.
